### PR TITLE
fix(huobi): watchBalance timestamp fix #19812

### LINKED ISSUE #19812: Huobi watchBalance results in undefined timestamp
### CCXT Version

4.1.36

### Description

The result of the function is an undefined timestamp and datetime when executing exchange.watchBalance()

I think it can be resolved by amending to below code in pro/huobi.ts:

### Code

```
  handleBalance (client: Client, message) {
        const channel = this.safeString (message, 'ch');
        this.balance['info'] = this.safeValue (message, 'data');
        if (channel !== undefined) {
            // spot balance
            const timestamp = this.safeInteger (message, 'changeTime');
            this.balance['timestamp'] = timestamp;
            this.balance['datetime'] = this.iso8601 (timestamp);
            const data = this.safeValue (message, 'data', {});
            const currencyId = this.safeString (data, 'currency');
            const code = this.safeCurrencyCode (currencyId);
            const account = this.account ();
            account['free'] = this.safeString (data, 'available');
            account['total'] = this.safeString (data, 'balance');
            this.balance[code] = account;
            this.balance = this.safeBalance (this.balance);
            client.resolve (this.balance, channel);
        } else {
            // contract balance
            const timestamp = this.safeInteger (message, 'ts');
            this.balance['timestamp'] = timestamp;
            this.balance['datetime'] = this.iso8601 (timestamp);
            const data = this.safeValue (message, 'data', []);
            const dataLength = data.length;
            if (dataLength === 0) {
                return;
            }
            const first = this.safeValue (data, 0, {});
            const topic = this.safeString (message, 'topic');


```

### PR DESCRIPTION
**Pull request by pcriadoperez**
- fix(huobi): watchBalance timestamp fix #19812

### CODE REVIEW

**Comment by pcriadoperez** [comment]
```js
{
  timestamp: 1699144483271,
  datetime: '2023-11-05T00:34:43.271Z',
  info: {
    currency: 'usdt',
    accountId: 44234548,
    balance: '115.096849200715990336',
    available: '104.296849200715990293',
    changeType: 'order.place',
    accountType: 'trade',
    seqNum: 598,
    changeTime: 1699144483271
  },
  USDT: { free: 104.296849200716, used: 10.8, total: 115.096849200716 },
  free: {
    USDT: 104.296849200716,
    BTC: 6.150257778e-8,
    DOGE: 0.08,
    XRP: 19.96,
    LTC: 0.000896272861189802,
    ADA: 4.5920051346985975
  },
  used: { USDT: 10.8, BTC: 0, DOGE: 0, XRP: 0, LTC: 0, ADA: 0 },
  total: {
    USDT: 115.096849200716,
    BTC: 6.150257778e-8,
    DOGE: 0.08,
    XRP: 19.96,
    LTC: 0.000896272861189802,
    ADA: 4.5920051346985975
  },
  BTC: { free: 6.150257778e-8, used: 0, total: 6.150257778e-8 },
  DOGE: { free: 0.08, used: 0, total: 0.08 },
  XRP: { free: 19.96, used: 0, total: 19.96 },
  LTC: { free: 0.000896272861189802, used: 0, total: 0.000896272861189802 },
  ADA: { free: 4.5920051346985975, used: 0, total: 4.5920051346985975 }
}
```
```python
huobi.watchBalance() (Timestamp is null as changetime is only shown when a change has happened)
{'ADA': {'free': 0.07789947610353354,
         'total': 0.07789947610353354,
         'used': 0.0},
 'BTC': {'free': 6.150257778e-08, 'total': 6.150257778e-08, 'used': 0.0},
 'DOGE': {'free': 0.08, 'total': 0.08, 'used': 0.0},
 'LTC': {'free': 0.000896272861189802,
         'total': 0.000896272861189802,
         'used': 0.0},
 'USDT': {'free': 13.000000008, 'total': 13.000000008, 'used': 1e-18},
 'XRP': {'free': 19.96, 'total': 19.96, 'used': 0.0},
 'datetime': None,
 'free': {'ADA': 0.07789947610353354,
          'BTC': 6.150257778e-08,
          'DOGE': 0.08,
          'LTC': 0.000896272861189802,
          'USDT': 13.000000008,
          'XRP': 19.96},
 'info': {'accountId': 48956839,
          'accountType': 'trade',
          'available': '0.000896272861189802',
          'balance': '0.000896272861189802',
          'changeTime': None,
          'changeType': None,
          'currency': 'ltc',
          'seqNum': 14},
 'timestamp': None,
 'total': {'ADA': 0.07789947610353354,
           'BTC': 6.150257778e-08,
           'DOGE': 0.08,
           'LTC': 0.000896272861189802,
           'USDT': 13.000000008,
           'XRP': 19.96},
 'used': {'ADA': 0.0,
          'BTC': 0.0,
          'DOGE': 0.0,
          'LTC': 0.0,
          'USDT': 1e-18,
          'XRP': 0.0}}
          ```

File: ts/src/krakenfutures.ts [hunk 1]
```diff
@@ -2142,6 +2139,7 @@ export default class krakenfutures extends Exchange {
         }
         const url = this.urls['api'][api] + query;
```
Comment (carlosmiei) [inline ts/src/krakenfutures.ts:2140]: @pcriadoperez I think something went wrong, this is the `krakenfutures` file 😅

Comment (sc0Vu) [reply ts/src/krakenfutures.ts:2140]: It looks like in another branch (https://github.com/ccxt/ccxt/pull/19836/files#diff-0f6be5c50c2d3c3beb8b6a55b945b3325e30ac7ed6085e3c7846b06fb28dc510)

**Comment by carlosmiei** [comment]
Yeah I believe this is the correct one: https://github.com/ccxt/ccxt/pull/19836

**Pull request closed without merging**